test/runtests.jl

```diff
@@ -25,19 +25,17 @@ include("test_utils.jl")
     println(" ")
     @info "Ran expected_loglik tests"
 
-    @testset "SparseVariationalApproximation" begin
```

Comment:
was it intentional to remove the `@testset`?

Reply:
Yes. We have test sets inside the files. We don't really need them at the top level.

Reply:
given that there's now a bunch of separate approximations in this package that each might be spread over multiple files, I had thought it makes sense to have "outer testsets" to make clear how they all fit together.... there isn't any limit on testset nesting is there? it's just to make it easy to figure out what goes wrong when it does

Reply:
Right.

My strong preference (which we used to follow consistently elsewhere, but isn't properly followed in AbstractGPs at the minute) is to have test sets in `runtests.jl` precisely mirror the directory structure of a project, and for there to be a 1-1 match between `src` and `test` files (see the [explanation](https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/blob/33d64d1425a4aa8748bd8ca453ca5be41eb53280/test/runtests.jl#L26) in KernelFunctions, which we should really copy + paste to all of our packages). The point being that if the tests ought to be grouped together in some sense, that's a sign that the code itself ought to be grouped together, either in the same file or multiple files in the same directory.

So my feeling is that we need to overhaul the structure here a bit (I find it weird that the elbo code is separated from the approximations, and I would much prefer to have the consistency tests co-located with the other tests for `SparseVariationalApproximation`). As a consequence, I've been a bit lax with how I've structured tests in this PR.

I'm happy to revert this particular change in favour of a proper overhaul in a future PR though.

Reply:
That makes a lot of sense, and I'd be happy to just turn it into an issue for now, and move them into their own subdirectory at a later point:) in that case it doesn't make much difference whether to revert or remove the outer testset

Reply:
Great. Issue opened https://github.com/JuliaGaussianProcesses/ApproximateGPs.jl/issues/75

I'll resolve this.